tools/ja-rule/ja-rule.cpp

```diff
@@ -0,0 +1,307 @@
+  }
+
+  ~InputHandler() {
+    m_stdin_handler.reset();
+  }
+
+  void DeviceEvent(USBDeviceManager::EventType event,
+                   OpenLightingDevice* device) {
+    if (event == USBDeviceManager::DEVICE_ADDED) {
+      OLA_INFO << "Open Lighting Device added";
+      m_device = device;
+      m_device->SetHandler(&m_handler);
+    } else {
+      OLA_INFO << "Open Lighting Device removed";
+      m_device = NULL;
+    }
+  }
+
+  void Input(int c) {
+    switch (c) {
+      case 'd':
+        SendDMX();
+        break;
+      case 'e':
+        SendEcho();
+        break;
+      case 'f':
+        GetFlags();
+        break;
+      case 'h':
+        PrintCommands();
+        break;
+      case 'l':
+        GetLogs();
+        break;
+      case 'q':
+        m_ss->Terminate();
+        break;
+      case 'w':
+        WriteLog();
+        break;
+      default:
+        {}
+    }
+  }
+
+  void PrintCommands() {
+    cout << "Commands:" << endl;
+    cout << " d - Send DMX frame" << endl;
+    cout << " e - Send Echo command" << endl;
+    cout << " f - Fetch Flags State" << endl;
+    cout << " h - Print this help message" << endl;
+    cout << " l - Fetch Logs" << endl;
+    cout << " w - Write Log" << endl;
+    cout << " q - Quit" << endl;
+  }
+
+ private:
+  SelectServer* m_ss;
+  auto_ptr<StdinHandler> m_stdin_handler;
+  MessageHandler m_handler;
+  OpenLightingDevice* m_device;
+  unsigned int m_log_count;
+
+  bool CheckForDevice() const {
+    if (!m_device) {
+      OLA_INFO << "Device not present";
+      return false;
+    }
+    return true;
+  }
+
+  void GetLogs() {
+    if (!CheckForDevice()) {
+      return;
+    }
+
+    m_device->SendMessage(OpenLightingDevice::GET_LOG, NULL, 0);
+  }
+
+  void GetFlags() {
+    if (!CheckForDevice()) {
+      return;
+    }
+
+    m_device->SendMessage(OpenLightingDevice::GET_FLAGS, NULL, 0);
+  }
+
+  void SendDMX() {
+    if (!CheckForDevice()) {
+      return;
+    }
+
+    const uint8_t payload[] = {255, 1, 2, 3, 4, 5, 6};
+    m_device->SendMessage(OpenLightingDevice::TX_DMX, payload,
+                          arraysize(payload));
+  }
+
+  void SendEcho() {
+    if (!CheckForDevice()) {
+      return;
+    }
+
+    const uint8_t payload[] = {'e', 'c', 'h', 'o', ' ', 't', 'e', 's', 't'};
+    m_device->SendMessage(OpenLightingDevice::ECHO_COMMAND, payload,
+                          arraysize(payload));
+  }
+
+  void WriteLog() {
+    if (!CheckForDevice()) {
+      return;
+    }
+
+    std::ostringstream str;
+    str << "Log Test " << m_log_count++ << ", this is quite long";
+    const string payload = str.str();
+    m_device->SendMessage(OpenLightingDevice::WRITE_LOG,
+                          reinterpret_cast<const uint8_t*>(payload.c_str()),
+                          payload.size());
+  }
+
+  DISALLOW_COPY_AND_ASSIGN(InputHandler);
+};
+
+/*
+ * Main.
+ */
+int main(int argc, char **argv) {
+  ola::AppInit(&argc, argv, "[ options ]", "Ja Rule Admin Tool");
```

Comment:
What flags? Right now there are no extra flags for this binary. AppInit takes care of parsing the flags.